packages/grant-explorer/src/features/api/utils.ts

```diff
@@ -420,3 +421,50 @@ export const listenForOutsideClicks = ({
     });
   };
 };
+
+export const useTokenPrice = (tokenId: string | undefined) => {
+  const [tokenPrice, setTokenPrice] = useState<number>();
+  const [error, setError] = useState<Response | undefined>();
+  const [loading, setLoading] = useState(false);
+
```

Comment:
imo I think checking for an empty token id is better than running through the request. 
```
  if (!tokenId) {
    setError(new Error('Token ID is not provided.'));
    return;
  }
```